File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
def dfs(numbers, target, depth, sum):
    if depth == len(numbers):
        if sum == target:
            return 1
        else:
            return 0
        
    return dfs(numbers, target, depth+1, sum+numbers[depth]) + dfs(numbers, target, depth+1, sum-numbers[depth])

def solution(numbers, target):
    return dfs(numbers, target, 0, 0)

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.java
class Solution {
    public int dfs(int[] numbers, int target, int depth, int sum){
        if(depth == numbers.length) {
            if(sum == target) return 1;
            else return 0;
        }
        
        // 현재 depth의 정수를 더해서 다음 depth의 dfs를 수행함.
        return dfs(numbers, target, depth+1, sum+numbers[depth]) + 
        // 현재 depth의 정수를 빼서 다음 depth의 dfs를 수행함.
        dfs(numbers, target, depth+1, sum-numbers[depth]);
    }
    
    public int solution(int[] numbers, int target) {
        return dfs(numbers, target, 0, 0);
    }
}